public/docs/_examples/cb-http-interception/ts/app/http-interceptor.ts

```diff
@@ -0,0 +1,127 @@
+/////// Experimental Generic HttpInterceptor ////////
+
+import { Injectable } from '@angular/core';
+
+import { Connection, ConnectionBackend, Http,
+         ReadyState, Request, RequestOptions,
+         Response, XHRBackend } from '@angular/http';
+
+import 'rxjs/add/observable/throw';
+
+import { of }         from 'rxjs/observable/of';
+import { Observable } from 'rxjs/Observable';
+import { Subject }    from 'rxjs/Subject';
+
+type interceptorFn = (act: HttpInterceptorActivity) => HttpInterceptorActivity;
+
+@Injectable()
+/**
+ * Http Backend Interceptor raises request and response activity events.
+ * Request interceptors can modify the request before it is sent (e.g, add headers) and can
+ * set the activity response to bypass XHR and return the response immediately.
+ * Response interceptors post-process responses before passing them to`http`.
+ */
+export class HttpInterceptor extends ConnectionBackend {
+
+  private requestInterceptors:  interceptorFn[] = [];
+  private responseInterceptors: interceptorFn[] = [];
+  private responseErrorInterceptors: interceptorFn[] = [];
+
+  private activitySubject = new Subject<HttpInterceptorActivity>();
+  activity = this.activitySubject.asObservable();
+
+  constructor(private xhrBackend: XHRBackend) { super(); }
+
+  // Todo: remove interceptors?
+
+  addRequestInterceptor(fn: interceptorFn) {
+    this.requestInterceptors.push(fn);
+  }
+
+  addResponseInterceptor(fn: interceptorFn) {
+    this.responseInterceptors.push(fn);
+  }
+
+  addResponseErrorInterceptor(fn: interceptorFn) {
+    this.responseInterceptors.push(fn);
```

Comment:
For anyone who is using this until HTTP Interceptors land... this line should read:
`this.responseErrorInterceptors.push(fn);`